osgi-dependencies/proxy/src/main/java/com/cognifide/aet/proxy/ProxyServerProvider.java

```diff
@@ -44,19 +44,20 @@
           unbind = "unbindProxyManager")
   private final Map<String, ProxyManager> collectorManagers = Maps.newConcurrentMap();
 
-  public ProxyServerWrapper createProxy(String useProxy, int port) throws ProxyException {
-    String proxyType = useProxy;
-    if ("true".equals(useProxy)) {
+  public ProxyServerWrapper createProxy(String usedProxy) throws ProxyException {
+    String proxyType = usedProxy;
+    if ("true".equals(usedProxy)) {
```

Comment:
There are two things I don't like

1. Passing boolean values as Strings. IMO, as soon as the value is read from the file, it should be converted to the most specific type possible in a given context.
2.  Mixing values with seemingly disparate semantics. 

Let's have a look at the current code.

    String proxyType;
    if ("true".equals(proxyTypeConfig)) {
        proxyType = DEFAULT_PROXY_MANAGER;
    } else {
        proxyType = proxyTypeConfig;
    }

It looks better now. `"true"` does not look like third alternative to `"embedded"` and `"rest`".

I'm not sure I understand the meaning of the name `proxyTypeConfig`. What exactly does it mean? That the default config should be used? If so, rename it to something along the lines of `useDefaultProxyType`. Can you show me an example of the XML file where this is configured?

On a side note, you could get rid of the null initialisation by converting the `if-else` block to a simple assignment:

  String proxyType = "true".equals(proxyTypeConfig) ? DEFAULT_PROXY_MANAGER : proxyTypeConfig;

Finally, if `proxyTypeConfig` is a boolean, the constructor should probably not take a `String` parameter.